python/langsmith/client.py

```diff
@@ -742,6 +746,14 @@ def _content_above_size(self, content_length: Optional[int]) -> Optional[str]:
         if size_limit is None:
             return None
         if content_length > size_limit:
+            should_debug_crash_dump = os.getenv("LANGSMITH_DEBUG_CRASH_DUMP") in [
+                "1",
+                "true",
+            ]
+            if should_debug_crash_dump and request is not None:
```

Comment:
This doesn't belong in a function that checks content length - it should go outside as the next step.

Also I think we would benefit from seeing other things in the dump even if the payload size doesn't surpass (since these are also of interest)

Reply:
Do you think we should always dump? Had intended this to only dump if we exceeded the content length limit (and also potentially dump here if other errors occur)

Reply:
Only on error, but this is intended for debugging connection errors, correct?

Reply:
Ah yes it's a connection error but specifically the one triggered by the content length limit.

> Failed to batch ingest runs: langsmith.utils.LangSmithConnectionError: Connection error caused failure to POST https://api.smith.langchain.com/runs/batch in LangSmith API. The content length of 270646872 bytes exceeds the maximum size limit of 20971520 bytes.

Their input and output tokens and traces seem to be pretty small, so we wanted to see what else was in their payload which was causing them to exceed 20mb.

We can also log more generically on ConnectionError though if you think that's better! It does make sense to log for other types of connection errors too